src/types/event.d.ts

```diff
@@ -3,15 +3,12 @@
 // https://github.com/promotedai/schema/blob/main/proto/event/event.proto
 // https://github.com/promotedai/schema/blob/main/proto/common/common.proto
 
-// TODO - auto-gen type from proto.
-export interface User {
-  userInfo?: UserInfo;
+export interface User extends HasUserInfo {
```

Comment:
This one in particular looks a little odd and made me wonder (not for the first time) why we don't use classes instead of interfaces for some of these. (I think if we used a proto-TS generate that's likely what we'd have.)

Reply:
I could see us switching to classes at some point if it enables something (e.g. lower latency or rpc sizes).  Right now, the client code looks cleaner to just use dictionaries.